### ISSUE TITLE
Implement pagination

### ISSUE THREAD

**Issue by nicogarcia** [nicogarcia/classes-web-client#18]
- Teachers result
- Student made reviews
- Teachers reviews

**Comment by nicogarcia** [comment]
Así es el pagination de materialize:
![image](https://cloud.githubusercontent.com/assets/241306/9790609/584602be-57ad-11e5-90cf-eb69a3f8f88f.png)

- ¿Deberíamos poner las flechitas dobles para ir al principio y final?
- ¿Deberíamos hacer collapse si hay muchas páginas? ¿Como funcionaría?

@federicojasson

**Comment by federicojasson** [comment]
¿No vienen implementadas esas características?
Yo creo que sí (tanto flechas dobles como establecer un máximo en la cantidad de números que muestra).

Otra cosa que puede ser importante para la usabilidad es poner el menú de paginación arriba y abajo de los resultados (me lo recomendaron una vez, para que el usuario sepa rápido que hay más resultados disponibles).

**Comment by nicogarcia** [comment]
Si, pero le faltan cosas a la directiva. Voy a hacerla a mano y después les muestro.

**Comment by tomasgil123** [comment]
En twitter, por ejemplo, no tenes pagination. Cuando llegas al fondo automaticamente se te cargan mas tweets. Asi evitas que el usuario tenga que hacer click.

**Comment by nicogarcia** [comment]
@federicojasson Es más complicado y tenes que cargar todo el contenido. Cuál era el problema de eso?

**Comment by federicojasson** [comment]
Muchos problemas:

- No tenés link a una página específica (no podés volver a donde estabas si, por ejemplo, te metiste en un perfil o te registraste en el sitio).

- Carga mucha información a la vez (lo cual insume mucha memoria). Esto es un problema especialmente en celulares. Basta con ver el ejemplo de Facebook: cuando ves fotos se te van acumulando y nunca se libera esa memoria (se ralentiza el navegador).

- Pueden quedar repeticiones de resultados si al momento de la búsqueda un profesor se registra (esto ya sí es poco probable).

**Comment by nicogarcia** [comment]
El mayor problema es el segundo, el primero con bastante trabajo se puede solucionar.

**Comment by tomasgil123** [comment]
Claro, lo costos superan por mucho a los beneficios. No tiene sentido hacerlo como en twitter
